File: common/src/main/java/org/sqlbuilder/common/ListCollector.java
package org.sqlbuilder.common;

import java.io.Closeable;
import java.util.ArrayList;

public class ListCollector<T extends SetId> extends ArrayList<T> implements Closeable
{
	private final boolean isOpen = false;

	private int allocator = 0;

	public ListCollector()
	{
		super();
	}

	public ListCollector<T> open()
	{
		return this;
	}

	@Override
	public boolean add(T item)
	{
		item.setId(++allocator);
		return super.add(item);
	}

	@Override
	public void close()
	{
		clear();
	}

	public String status()
	{
		return ":" + size();
	}
}

File: common/src/main/java/org/semantikos/common/ListCollector.kt
package org.semantikos.common

import java.io.Closeable

class ListCollector<T : SetId> : ArrayList<T>(), Closeable {

    private var allocator = 0

    fun open(): ListCollector<T> {
        return this
    }

    override fun add(item: T): Boolean {
        item.setId(++allocator)
        return super.add(item)
    }

    override fun close() {
        clear()
    }

    fun status(): String {
        return ":$size"
    }
}
